cumulus/pallets/collator-selection/src/migration.rs

```diff
@@ -90,3 +196,54 @@ pub mod v1 {
 		}
 	}
 }
+
+#[cfg(all(feature = "try-runtime", test))]
+mod tests {
+	use super::*;
+	use crate::{
+		migration::v2::Candidates,
+		mock::{new_test_ext, Balances, Test},
+	};
+	use frame_support::{
+		traits::{Currency, ReservableCurrency, StorageVersion},
+		BoundedVec,
+	};
+	use sp_runtime::traits::ConstU32;
+
+	#[test]
+	fn migrate_to_v2() {
+		new_test_ext().execute_with(|| {
+			let storage_version = StorageVersion::new(1);
+			storage_version.put::<Pallet<Test>>();
+
+			let who = 1u64;
+
+			// Set balance to 100
+			Balances::make_free_balance_be(&who, 100u64);
+			// Reserve 20 -> 10 for the "old" candidacy and 10 for the "new"
+			Balances::reserve(&who, 20u64).unwrap();
+			// Candidate info
+			let candidate = CandidateInfo { who, deposit: 10u64 };
+			let bounded_candidates =
```

Comment:
Let's also add 2 more candidates with the same sanity checks:
- one present only in `Candidates`
- one present only in `CandidateList`